infrastructure/src/main/java/org/corfudb/infrastructure/BatchProcessor.java

```diff
@@ -59,7 +59,7 @@ public class BatchProcessor implements AutoCloseable {
      * This is persisted in the ServerContext by the LogUnitServer to withstand restarts.
      */
     @Getter
-    private long sealEpoch;
+    private volatile long sealEpoch;
```

Comment:
Note: the volatile keyword is added here since sealEpoch is also accessed by the LogMetadataRecorder thread.